node/rpc/src/chain_api.rs

```diff
@@ -80,3 +94,77 @@ pub(crate) async fn chain_block_messages<DB: BlockStore + Send + Sync + 'static>
     };
     Ok(ret)
 }
+
+pub(crate) async fn chain_get_tipset_by_height<DB: BlockStore + Send + Sync + 'static>(
+    data: Data<State<DB>>,
+    Params(params): Params<(ChainEpoch, TipsetKeys)>,
+) -> Result<TipsetJson, JsonRpcError> {
+    let height = params.0;
+    let tsk = params.1;
+    let ts = chain::tipset_from_keys(data.store.as_ref(), &tsk)?;
+    let tss = chain::tipset_by_height(data.store.as_ref(), height, ts, true)?;
+    Ok(TipsetJson(tss))
+}
+
+pub(crate) async fn chain_get_genesis<DB: BlockStore + Send + Sync + 'static>(
+    data: Data<State<DB>>,
+) -> Result<Option<TipsetJson>, JsonRpcError> {
+    let genesis = chain::genesis(data.store.as_ref())?.ok_or("can't find genesis tipset")?;
+    let gen_ts = Tipset::new(vec![genesis])?;
+    Ok(Some(TipsetJson(gen_ts)))
+}
+
+pub(crate) async fn chain_head<DB: BlockStore + Send + Sync + 'static>(
+    data: Data<State<DB>>,
+) -> Result<TipsetJson, JsonRpcError> {
+    let heaviest =
+        chain::get_heaviest_tipset(data.store.as_ref())?.ok_or("can't find heaviest tipset")?;
+    Ok(TipsetJson(heaviest))
+}
+
+pub(crate) async fn chain_tipset_weight<DB: BlockStore + Send + Sync + 'static>(
+    data: Data<State<DB>>,
+    Params(params): Params<(TipsetKeys,)>,
+) -> Result<String, JsonRpcError> {
+    let tsk = params.0;
```

Comment:
```suggestion
    let (tsk,) = params;
```
Optional for this one, but just showing these could have the same format if it makes it easier in the future